starter-code/src/functions-and-arrays.js

```diff
@@ -75,8 +156,36 @@ const wordsCount = [
   'matter'
 ];
 
+function howManyTimes(arr,word) {
+  if (arr.length === 0){
+    return 0
+  }
+  let counter = 0
+    for (i=0; i < arr.length; i++) {
+      if (arr[i] === word) {
+        counter++
+      } 
+    } return counter
+  }
+  howManyTimes(words, 'matter')
+
 // Bonus
 
+function greatestProduct(){
```

Comment:
You're definitely on the right track! If you have a chance to go back to this function, you'll end up having to make similar loops to look for vertical and diagonal combinations. It's very redundant, so don't worry too much about it!